### PR TITLE
BZ1761297 - added storage in use protection

### PR DESCRIPTION
**Pull request by bobfuru**
Adding content from upstream, based on [BZ 1761297](https://bugzilla.redhat.com/show_bug.cgi?id=1761297).
@liangxia PTAL for QE
@bergerhoffer or @lamek PTAL for peer review

### CODE REVIEW

File: modules/storage-persistent-storage-lifecycle.adoc [hunk 1]
```diff
@@ -57,9 +57,18 @@ PVs by including `persistentVolumeClaim` in the Pod's volumes block.
 ifdef::openshift-origin,openshift-enterprise[]
 
 [id="pvcprotection_{context}"]
-== PVC protection
+== Storage Object in Use Protection
 
-PVC protection is enabled by default.
+The Storage Object in Use Protection feature ensures that PVCs in active use by a Pod and PVs that are bound to PVCs are not removed from the system, as this may result in data loss.
```
Comment (kalexand-rh) [inline modules/storage-persistent-storage-lifecycle.adoc]: You need to go up and expand PV and PVC on first use.
s/may/can

File: modules/storage-persistent-storage-lifecycle.adoc [hunk 2]
```diff
@@ -57,9 +57,18 @@ PVs by including `persistentVolumeClaim` in the Pod's volumes block.
...
+
+Storage Object in Use Protection is enabled by default.
+
+[NOTE]
+====
+A PVC is in active use by a Pod when a Pod object exists that is using the PVC.
```
Comment (kalexand-rh) [inline modules/storage-persistent-storage-lifecycle.adoc]: s/is using/uses

File: modules/storage-persistent-storage-lifecycle.adoc [hunk 3]
```diff
@@ -57,9 +57,18 @@ PVs by including `persistentVolumeClaim` in the Pod's volumes block.
...
+====
+
+If a user deletes a PVC in active use by a Pod, the PVC is not removed immediately. PVC removal is postponed until the PVC is no longer actively used by any Pods. Also, if a cluster admin deletes a PV that is bound to a PVC, the PV is not removed immediately. PV removal is postponed until the PV is no longer bound to a PVC.
```
Comment (kalexand-rh) [inline modules/storage-persistent-storage-lifecycle.adoc]: s/in active/that is in active
s/immediately./immediately,

**Review by kalexand-rh — COMMENTED** [review]
couple of picks, then it LGTM

**Comment by bobfuru** [comment]
> The preview will be available shortly at:
> 
> * https://BZ1761297--ocpdocs.netlify.com/
Updated section:
https://bz1761297--ocpdocs.netlify.com/openshift-enterprise/latest/storage/understanding-persistent-storage.html#pvcprotection_understanding-persistent-storage

File: modules/storage-persistent-storage-lifecycle.adoc (hunk 1 above)
Comment (bobfuru) [reply modules/storage-persistent-storage-lifecycle.adoc]: PV and PVC are expanded in first use of the [assembly](https://docs.openshift.com/container-platform/4.2/storage/understanding-persistent-storage.html#provisioning_understanding-persistent-storage). Elsewhere, h2s are using the initialisms. LMK if that's not correct.

Comment (kalexand-rh) [reply modules/storage-persistent-storage-lifecycle.adoc]: If each module is meant to be standalone, there's no guarantee that first use will be elsewhere.

File: modules/storage-persistent-storage-lifecycle.adoc (hunk 3 above)
Comment (bobfuru) [reply modules/storage-persistent-storage-lifecycle.adoc]: Leaving `immediately.` to avoid run-on sentence.

**Comment by lamek** [comment]
LGTM too :-)

**Comment by liangxia** [comment]
LGTM

**Pull request merged**